addons/storyshots/storyshots-core/src/test-bodies.js

```diff
@@ -1,12 +1,15 @@
 import 'jest-specific-snapshot';
 
+const isFunction = obj => !!(obj && obj.constructor && obj.call && obj.apply);
+const optionsOrCallOptions = (opts, story) => (isFunction(opts) ? opts(story) : opts);
```

Comment:
Why is it better than just `typeof opts === 'function'` ?

Reply:
Basically just the function underscore uses. Preferred taking something that worked at-scale than going with the naive implementation (from what I read at some cases objects can return 'function').